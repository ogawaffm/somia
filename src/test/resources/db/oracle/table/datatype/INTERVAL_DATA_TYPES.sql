BEGIN
    EXECUTE IMMEDIATE 'DROP TABLE TEST.INTERVAL_DATA_TYPES PURGE';
EXCEPTION
    WHEN OTHERS THEN
        IF SQLCODE != -942 THEN
            RAISE;
        END IF;
END;
/
CREATE TABLE TEST.INTERVAL_DATA_TYPES
(
    INTERVAL_YEAR_TO_MONTH_COL     INTERVAL YEAR TO MONTH,
    INTERVAL_YEAR_TO_MONTH_1_COL   INTERVAL YEAR(1) TO MONTH,
    INTERVAL_YEAR_TO_MONTH_2_COL   INTERVAL YEAR(2) TO MONTH,
    INTERVAL_YEAR_TO_MONTH_9_COL   INTERVAL YEAR(9) TO MONTH,

    INTERVAL_DAY_TO_SECOND_COL     INTERVAL DAY TO SECOND,

    INTERVAL_DAY_0_TO_SECOND_COL   INTERVAL DAY(0) TO SECOND,
    INTERVAL_DAY_2_TO_SECOND_COL   INTERVAL DAY(2) TO SECOND,
    INTERVAL_DAY_9_TO_SECOND_COL   INTERVAL DAY(9) TO SECOND,

    INTERVAL_DAY_TO_SECOND_0_COL   INTERVAL DAY TO SECOND(0),
    INTERVAL_DAY_TO_SECOND_6_COL   INTERVAL DAY TO SECOND(6),
    INTERVAL_DAY_TO_SECOND_9_COL   INTERVAL DAY TO SECOND(9),

    INTERVAL_DAY_0_TO_SECOND_0_COL INTERVAL DAY(0) TO SECOND(0),
    INTERVAL_DAY_0_TO_SECOND_6_COL INTERVAL DAY(0) TO SECOND(6),
    INTERVAL_DAY_0_TO_SECOND_9_COL INTERVAL DAY(0) TO SECOND(9),

    INTERVAL_DAY_2_TO_SECOND_0_COL INTERVAL DAY(2) TO SECOND(0),
    INTERVAL_DAY_2_TO_SECOND_6_COL INTERVAL DAY(2) TO SECOND(6),
    INTERVAL_DAY_2_TO_SECOND_9_COL INTERVAL DAY(2) TO SECOND(9),

    INTERVAL_DAY_9_TO_SECOND_0_COL INTERVAL DAY(9) TO SECOND(0),
    INTERVAL_DAY_9_TO_SECOND_6_COL INTERVAL DAY(9) TO SECOND(6),
    INTERVAL_DAY_9_TO_SECOND_9_COL INTERVAL DAY(9) TO SECOND(9)

)