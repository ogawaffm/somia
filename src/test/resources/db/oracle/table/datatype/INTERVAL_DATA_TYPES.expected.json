{
  "name": "INTERVAL_DATA_TYPES",
  "catalogName": "",
  "schemaName": "TEST",
  "typeName": "TABLE",
  "comment": "",
  "columns": [
    {
      "name": "INTERVAL_YEAR_TO_MONTH_COL",
      "type": {
        "name": "INTERVAL YEAR(2) TO MONTH",
        "typeNumber": -103,
        "className": "java.lang.Object",
        "precision": 2,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 0,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "INTERVAL_YEAR_TO_MONTH_1_COL",
      "type": {
        "name": "INTERVAL YEAR(1) TO MONTH",
        "typeNumber": -103,
        "className": "java.lang.Object",
        "precision": 1,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 0,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "INTERVAL_YEAR_TO_MONTH_2_COL",
      "type": {
        "name": "INTERVAL YEAR(2) TO MONTH",
        "typeNumber": -103,
        "className": "java.lang.Object",
        "precision": 2,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 0,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "INTERVAL_YEAR_TO_MONTH_9_COL",
      "type": {
        "name": "INTERVAL YEAR(9) TO MONTH",
        "typeNumber": -103,
        "className": "java.lang.Object",
        "precision": 9,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 0,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "INTERVAL_DAY_TO_SECOND_COL",
      "type": {
        "name": "INTERVAL DAY(2) TO SECOND(6)",
        "typeNumber": -104,
        "className": "java.lang.Object",
        "precision": 2,
        "scale": 6,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 0,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "INTERVAL_DAY_0_TO_SECOND_COL",
      "type": {
        "name": "INTERVAL DAY(0) TO SECOND(6)",
        "typeNumber": -104,
        "className": "java.lang.Object",
        "precision": 0,
        "scale": 6,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 0,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "INTERVAL_DAY_2_TO_SECOND_COL",
      "type": {
        "name": "INTERVAL DAY(2) TO SECOND(6)",
        "typeNumber": -104,
        "className": "java.lang.Object",
        "precision": 2,
        "scale": 6,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 0,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "INTERVAL_DAY_9_TO_SECOND_COL",
      "type": {
        "name": "INTERVAL DAY(9) TO SECOND(6)",
        "typeNumber": -104,
        "className": "java.lang.Object",
        "precision": 9,
        "scale": 6,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 0,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "INTERVAL_DAY_TO_SECOND_0_COL",
      "type": {
        "name": "INTERVAL DAY(2) TO SECOND(0)",
        "typeNumber": -104,
        "className": "java.lang.Object",
        "precision": 2,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 0,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "INTERVAL_DAY_TO_SECOND_6_COL",
      "type": {
        "name": "INTERVAL DAY(2) TO SECOND(6)",
        "typeNumber": -104,
        "className": "java.lang.Object",
        "precision": 2,
        "scale": 6,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 0,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "INTERVAL_DAY_TO_SECOND_9_COL",
      "type": {
        "name": "INTERVAL DAY(2) TO SECOND(9)",
        "typeNumber": -104,
        "className": "java.lang.Object",
        "precision": 2,
        "scale": 9,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 0,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "INTERVAL_DAY_0_TO_SECOND_0_COL",
      "type": {
        "name": "INTERVAL DAY(0) TO SECOND(0)",
        "typeNumber": -104,
        "className": "java.lang.Object",
        "precision": 0,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 0,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "INTERVAL_DAY_0_TO_SECOND_6_COL",
      "type": {
        "name": "INTERVAL DAY(0) TO SECOND(6)",
        "typeNumber": -104,
        "className": "java.lang.Object",
        "precision": 0,
        "scale": 6,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 0,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "INTERVAL_DAY_0_TO_SECOND_9_COL",
      "type": {
        "name": "INTERVAL DAY(0) TO SECOND(9)",
        "typeNumber": -104,
        "className": "java.lang.Object",
        "precision": 0,
        "scale": 9,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 0,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "INTERVAL_DAY_2_TO_SECOND_0_COL",
      "type": {
        "name": "INTERVAL DAY(2) TO SECOND(0)",
        "typeNumber": -104,
        "className": "java.lang.Object",
        "precision": 2,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 0,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "INTERVAL_DAY_2_TO_SECOND_6_COL",
      "type": {
        "name": "INTERVAL DAY(2) TO SECOND(6)",
        "typeNumber": -104,
        "className": "java.lang.Object",
        "precision": 2,
        "scale": 6,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 0,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "INTERVAL_DAY_2_TO_SECOND_9_COL",
      "type": {
        "name": "INTERVAL DAY(2) TO SECOND(9)",
        "typeNumber": -104,
        "className": "java.lang.Object",
        "precision": 2,
        "scale": 9,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 0,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "INTERVAL_DAY_9_TO_SECOND_0_COL",
      "type": {
        "name": "INTERVAL DAY(9) TO SECOND(0)",
        "typeNumber": -104,
        "className": "java.lang.Object",
        "precision": 9,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 0,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "INTERVAL_DAY_9_TO_SECOND_6_COL",
      "type": {
        "name": "INTERVAL DAY(9) TO SECOND(6)",
        "typeNumber": -104,
        "className": "java.lang.Object",
        "precision": 9,
        "scale": 6,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 0,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "INTERVAL_DAY_9_TO_SECOND_9_COL",
      "type": {
        "name": "INTERVAL DAY(9) TO SECOND(9)",
        "typeNumber": -104,
        "className": "java.lang.Object",
        "precision": 9,
        "scale": 9,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 0,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    }
  ],
  "indexes": [
    {
      "name": "",
      "columns": [],
      "isUnique": true,
      "comment": "",
      "qualifierName": "",
      "isClustered": false,
      "isHashed": false,
      "isStatistic": true,
      "whereClause": "",
      "cardinality": 0,
      "pages": 0
    }
  ],
  "collation": "",
  "primaryKey": null,
  "bestRowIdentifier": null
}